pywbemtools/pywbemcli/_cmd_listener.py

```diff
@@ -0,0 +1,473 @@
+# (C) Copyright 2021 IBM Corp.
+# (C) Copyright 2021 Inova Development Inc.
+# All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Click Command definition for the listener command group which includes
+cmds to manage indication listeners that are separate invocations of
+pywbemcli representing a listener.
+
+NOTE: Commands are ordered in help display by their order in this file.
+"""
+
+from __future__ import absolute_import, print_function
+
+import subprocess
+import signal
+import argparse
+from time import sleep
+import click
+import psutil
+
+from pywbem import WBEMListener
+
+from .pywbemcli import cli
+from ._common import CMD_OPTS_TXT, GENERAL_OPTS_TXT, SUBCMD_HELP_TXT, \
+    format_table
+from ._common_options import add_options, help_option
+from ._click_extensions import PywbemcliGroup, PywbemcliCommand
+
+DEFAULT_LISTENER_PORT = 5988
+DEFAULT_LISTENER_PROTOCOL = 'http'
+
+LISTEN_OPTIONS = [
+    click.option('--port', type=int, metavar='PORT',
+                 required=False, default=DEFAULT_LISTENER_PORT,
+                 help=u'The port number listened on. '
+                 'Default: {}'.format(DEFAULT_LISTENER_PORT)),
+    click.option('--protocol', type=click.Choice(['http', 'https']),
+                 metavar='PROTOCOL',
+                 required=False, default=DEFAULT_LISTENER_PROTOCOL,
+                 help=u'The protocol used by the listener (http, https). '
+                 'Default: {}'.format(DEFAULT_LISTENER_PROTOCOL)),
+    click.option('--certfile', type=str, metavar='FILE',
+                 required=False, default=None,
+                 help=u'Path name of a PEM file containing the certificate '
+                 'that will be presented as a server certificate during '
```

Comment:
This one is going to be fun.  We will have to clarify the server cert, etc. since in this case the listener is the server and the indication exporter is the client.  Note that we will eventually have to make this more than just a certificate since this really requires a certificate trust store.

Reply:
On listener assuming server role in HTTPS: That's why I explained that this is presented by the listener as a server certificate. I agree that this requires some more words, but that would be in a security section of the two be written documentation for the listener command group.
DEFERRED to future update of this PR.

On certificate trust store:  The underlying WBEMListener class takes exactly these two arguments, and it directly passes them on to `ssl.SSLContext.load_cert_chain()`. We need to discuss this. Added it to the discussion items.
DEFERRED to future update of this PR.